Utilities/VirtualMemory.cpp

```diff
@@ -15,27 +16,30 @@ namespace memory_helper
 	void* reserve_memory(size_t size)
 	{
 #ifdef _WIN32
-		return VirtualAlloc(NULL, size, MEM_RESERVE, PAGE_NOACCESS);
+		void* ret = VirtualAlloc(NULL, size, MEM_RESERVE, PAGE_NOACCESS);
+		CHECK_ASSERTION(ret == NULL);
```

Comment:
CHECK_ASSERTION accepts "right" condition, in your code it's inverted everywhere.
Plus many return conditions are actually wrong.